pallets/credit/src/lib.rs

```diff
@@ -371,14 +373,20 @@ pub mod pallet {
             }
         }
 
-        fn update_credit_history(account_id: &T::AccountId, current_era: EraIndex) -> Weight {
-            let mut user_credit_history = Self::user_credit_history(&account_id);
+        pub fn update_credit_history(account_id: &T::AccountId, current_era: EraIndex) -> Weight {
+            let user_credit_data = Self::user_credit(&account_id).unwrap();
             let mut weight = T::DbWeight::get().reads_writes(1, 0);
+
+            if Self::user_credit_history(&account_id).is_empty() {
+                weight = weight
+                    .saturating_add(Self::init_credit_history(&account_id, user_credit_data.clone()));
+            }
+            let mut user_credit_history = Self::user_credit_history(&account_id);
+            weight = weight.saturating_add(T::DbWeight::get().reads_writes(1, 0));
+
             if !user_credit_history.is_empty() {
```

Comment:
this if not needed

Reply:
Yes, it is still reserved for reasons of prudent consideration.